libs/dh/esett/feature-outgoing-messages/src/lib/dh-outgoing-messages.component.ts

```diff
@@ -155,4 +158,35 @@ export class DhOutgoingMessagesComponent implements OnInit, OnDestroy {
   handlePageEvent({ pageIndex, pageSize }: PageEvent): void {
     this.pageMetaData$.next({ pageIndex, pageSize });
   }
+
+  download(): void {
+    if (!this.tableDataSource.sort) {
+      return;
+    }
+
+    const dataSorted = this.tableDataSource.sortData(
+      this.tableDataSource.filteredData,
+      this.tableDataSource.sort
+    );
+
+    const outgoingMessagesPath = 'eSett.outgoingMessages';
+
+    const headers = [
+      `"${translate(outgoingMessagesPath + '.columns.id')}"`,
+      `"${translate(outgoingMessagesPath + '.columns.calculationType')}"`,
+      `"${translate(outgoingMessagesPath + '.columns.messageType')}"`,
```

Comment:
Suggestion for an Frontend guild talk. Maybe we should look into having a exportToCsv baked into the watt-table. It already have all the data?

Reply:
Translations might be a challenge but it's worth exploring.